man/mkinitcpio.8.txt

```diff
@@ -100,6 +100,19 @@ Options
 *-z, \--compress* 'compress'::
 	Override the compression method with the 'compress' program.
 
+Environment
+-----------
+The following environment variables influence the program behavior:
```

Comment:
But the whole point was to make this enabled by default, without the need to set SOURCE_DATE_EPOCH.

Reply:
As per your earlier comment:

> The easy route of assuming SOURCE_DATE_EPOCH=0, or the slightly less easy route of assuming SOURCE_DATE_EPOCH=$(time_of_mkinitcpio_build)

The former can produce reproducible builds by default. If we go with that approach, we don't really need SOURCE_DATE_EPOCH at all. If no-one cares about timestamps in initramfs (I don't know of a reason to care about them) then we can just do that.

If we want the default to be the time of the mkinitcpio build (latest commit has this behaviour), then the default behaviour is not a reproducible build by definition right, if you run it twice you'll have different embedded timestamps.

Reply:
To distill this: is there any reason to care about timestamps at all here, if not, let's just touch -hcd @1 or whatever, strip all references to S_D_E and be done with it?

Reply:
That's what I was trying to get at, but I wrote this hastily and my intent wasn't clear. I suggested something like the build date of mkinitcpio rather than 0 for two reasons:

1) lack of user surprise. This is wishy washy, but some people might be oddly concerned about timestamps from 1970.
2) encoding the build timestamp might give a hint as to what version of mkinitcpio created the archive. However, that's rather redundant -- we already have a VERSION file in the cpio.

I guess I really don't care what the value is -- I just don't see the point of making this some sort of opt-in feature. If we can make the initramfs reproducible in the common case without side effects, I see no reason not to.

So yes, IMO, strip everything related to SOURCE_DATE_EPOCH and document the fact that mkinitcpio tries to make reproducible archives by default. You can call out compression as something that might make the archive *not* reproducible and mention "known good recipes".

Reply:
Gotcha. I hadn't realised you meant the build date of mkinitcpio itself rather than the runtime date.